File: core/src/main/java/com/cc/core/wechat/invoke/OpenUrlAction.java
package com.cc.core.wechat.invoke;

import android.content.Intent;

import com.cc.core.MyApplication;
import com.cc.core.actions.Action;
import com.cc.core.actions.ActionResult;
import com.cc.core.wechat.Wechat;

public class OpenUrlAction implements Action {
    @Override
    public ActionResult execute(Object... args) {
        Intent i = new Intent();
        i.setClassName(MyApplication.application(), Wechat.Resources.WEBVIEW_ACTIVITY_CLASS_NAME);
        i.putExtra("rawUrl", args[0].toString());
        i.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
        MyApplication.application().startActivity(i);
        return ActionResult.successResult();
    }

    @Override
    public String key() {
        return "wechat:open_url";
    }
}

File: core/src/main/java/com/cc/core/wechat/invoke/OpenUrlAction.kt
package com.cc.core.wechat.invoke

import android.app.Application
import android.content.Intent
import com.cc.core.ApplicationContext
import com.cc.core.actions.Action
import com.cc.core.actions.ActionResult
import com.cc.core.wechat.Wechat

class OpenUrlAction : Action {
    override fun execute(actionId : String, vararg args: Any?): ActionResult? {
        val i = Intent()
        i.setClassName(ApplicationContext.application(), Wechat.Resources.WEBVIEW_ACTIVITY_CLASS_NAME)
        i.putExtra("rawUrl", args[0].toString())
        i.flags = Intent.FLAG_ACTIVITY_NEW_TASK
        (ApplicationContext.application() as Application).startActivity(i)
        return ActionResult.successResult(actionId)
    }

    override fun key(): String? {
        return "wechat:open_url"
    }
}